### ISSUE TITLE
`merge_xml_files` does not work above python3.9

### ISSUE THREAD

**Issue by keiyamamo** [KVSlab/turtleFSI#46]
Since `getchildren()` is deprecated from python3.9, `merge_xml_files` does not work above python3.9
https://github.com/KVSlab/turtleFSI/blob/b4df3739fabfa7b9592ba9511a8b7fa464e9860f/turtleFSI/problems/__init__.py#L422
I modified the function and it works with python3.10 on my laptop, but I do not know if it works with python3.8 or lower. 
Can @johannesring  or @jorgensd  maybe test if replacing the `merge_xml_files` as follows works fine with python3.8?
You can use Taylor green problem that I mentioned in the other issue to test `merge_xml_files`.  
Here is the link to the branch https://github.com/keiyamamo/turtleFSI/tree/fix_save_viz
and also new `mega_xml_file` (in case you just want to copy this function).

```
def merge_xml_files(files):
    # Get first timestep and trees
    first_timesteps = []
    trees = []
    for f in files:
        trees.append(ET.parse(f))
        root = trees[-1].getroot()
        first_timesteps.append(float(root[0][0][0][2].attrib["Value"]))

    # Index valued sort (bypass numpy dependency)
    first_timestep_sorted = sorted(first_timesteps)
    indexes = [first_timesteps.index(i) for i in first_timestep_sorted]

    # Get last timestep of first tree
    base_tree = trees[indexes[0]]
    last_node = base_tree.getroot()[0][0][-1]
    ind = 2 if len(last_node.findall('*')) == 4 else 1
    last_timestep = float(last_node[ind].attrib["Value"])

    # Append
    for index in indexes[1:]:
        tree = trees[index]
        for node in tree.getroot()[0][0].findall('*'):
            ind = 2 if len(node.findall('*')) == 4 else 1
            if last_timestep < float(node[ind].attrib["Value"]):
                base_tree.getroot()[0][0].append(node)
                last_timestep = float(node[ind].attrib["Value"])

    # Seperate xdmf files
    new_file = [f for f in files if "_" not in f.name.__str__()]
    old_files = [f for f in files if "_" in f.name.__str__()]

    # Write new xdmf file
    base_tree.write(new_file[0], xml_declaration=True)

    # Delete xdmf file
    [f.unlink() for f in old_files]
```

**Comment by keiyamamo** [comment]
Right now, `environment.yml` specifies python version to be 3.10, but I do not know if we need python 3.10. 
We can downgrade python version to, for example, 3.8 and change the documentation to say that turtle works with python<3.9 instead of updating this function. 

https://github.com/KVSlab/turtleFSI/blob/b4df3739fabfa7b9592ba9511a8b7fa464e9860f/environment.yml#L7

What do you think, @johannesring @jorgensd ?

**Comment by jorgensd** [comment]
> Right now, `environment.yml` specifies python version to be 3.10, but I do not know if we need python 3.10. We can downgrade python version to, for example, 3.8 and change the documentation to say that turtle works with python<3.9 instead of updating this function.
> 
> https://github.com/KVSlab/turtleFSI/blob/b4df3739fabfa7b9592ba9511a8b7fa464e9860f/environment.yml#L7
> 
> What do you think, @johannesring @jorgensd ?

Python 3.7 is EOL (end of life) in June, so I wouldn't bother supporting it.
However, we should support Python3.8. 
Thus I've made #47 that only enforces a min bound on the python env, and we can test it with anaconda.

**Comment by johannesring** [comment]
I ran into the same problem in Oasis not long ago and fixed it by using `list(element)` instead of `element.getchildren()` as written in the [documentation](https://docs.python.org/2/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.getchildren). For turtleFSI this would be:
``` patch
--- a/turtleFSI/problems/__init__.py
+++ b/turtleFSI/problems/__init__.py
@@ -419,14 +419,14 @@ def merge_xml_files(files):
     # Get last timestep of first tree
     base_tree = trees[indexes[0]]
     last_node = base_tree.getroot()[0][0][-1]
-    ind = 1 if len(last_node.getchildren()) == 3 else 2
+    ind = 1 if len(list(last_node) == 3 else 2
     last_timestep = float(last_node[ind].attrib["Value"])
 
     # Append
     for index in indexes[1:]:
         tree = trees[index]
-        for node in tree.getroot()[0][0].getchildren():
-            ind = 1 if len(node.getchildren()) == 3 else 2
+        for node in list(tree.getroot()[0][0]):
+            ind = 1 if len(list(node) == 3 else 2
             if last_timestep < float(node[ind].attrib["Value"]):
                 base_tree.getroot()[0][0].append(node)
                 last_timestep = float(node[ind].attrib["Value"])
```

**Comment by keiyamamo** [comment]
solved #50
